packages/core/src/Mailbox.ts

```diff
@@ -24,6 +24,22 @@ export class Mailbox<T> {
     }
   }
 
+  // TODO: rethink this design
+  public prepend(event: T): void {
```

Comment:
Why?
1. to process the start signal **first** in the `ObservableActorRef`: https://github.com/statelyai/xstate/blob/76d738f0f8df6f6782a9b6d9b9bd565b4f6e1627/packages/core/src/ObservableActorRef.ts#L32
2. the mailbox can receive and queue events before the actor gets started
3. but they were ignored here because the inner service was not yet created (because that requires a start signal): https://github.com/statelyai/xstate/blob/76d738f0f8df6f6782a9b6d9b9bd565b4f6e1627/packages/core/src/actors.ts#L431

Reply:
In the future, we can model this as a priority queue, with signals having the highest priority.

This will also help in situations where a mailbox has a lot of events in queue but it needs to be shut down ASAP. The "stop signal" should take priority.

Reply:
> This will also help in situations where a mailbox has a lot of events in queue but it needs to be shut down ASAP. The "stop signal" should take priority.

This particular case is currently handled like this:
https://github.com/statelyai/xstate/blob/76d738f0f8df6f6782a9b6d9b9bd565b4f6e1627/packages/core/src/Mailbox.ts#L18-L25

and I feel like it's a good solution here as we want to drop all queued items when receiving a stop signal anyway.

The idea of priorities etc is definitely intriguing but I think other use cases would have to back up its implementation (and if we'd implement such a thing then stop signal maybe could be handled using this mechanism).